File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/RemoteDeposit/ProfitStars.cs
namespace Psi.Data.Models.ClientConfigurationModels.RemoteDeposit
{
    public class ProfitStars : SettingsBaseHelper
    {
        public ProfitStars(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("RemoteDeposit.ProfitStars.StoreId")]
        public long StoreId
        {
            get => GetLongValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.EntityId")]
        public int EntityId
        {
            get => GetIntValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.StoreKey")]
        public string StoreKey
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.SharedSecret")]
        public string SharedSecret
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.FiIdentifier")]
        public string FiIdentifier
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.AdminServiceUrl")]
        public string AdminServiceUrl
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.ServiceUrl")]
        public string ServiceUrl
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.VendorName")]
        public string VendorName
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.AndroidAppBundleId")]
        public string AndroidAppBundleId
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.IosAppBundleId")]
        public string IosAppBundleId
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("RemoteDeposit.ProfitStars.SyncAccounts.Enabled")]
        public bool SyncAccountsEnabled
        {
            get => GetBoolValue();
            set => SetValue(value);
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/Psi/Data/Models/ClientConfigurationModels/RemoteDeposit/ProfitStars.ts
import { Setting, ISettingsGroup, ISettingsMetadata } from '@models/base/types';
export interface ProfitStarsConfig {
    StoreId: number;
    EntityId: number;
    StoreKey: string;
    SharedSecret: string;
    FiIdentifier: string;
    AdminServiceUrl: string;
    ServiceUrl: string;
    VendorName: string;
    AndroidAppBundleId: string;
    IosAppBundleId: string;
    SyncAccountsEnabled: boolean;
}

export class ProfitStars implements ISettingsGroup {
    static readonly metadata: ISettingsMetadata = {
        groupName: 'ProfitStars'
    };


            private _storeId: number;
            get storeId(): number {
                return this._storeId;
            }
            set storeId(value: number) {
                this._storeId = value;
            }

            private _entityId: number;
            get entityId(): number {
                return this._entityId;
            }
            set entityId(value: number) {
                this._entityId = value;
            }

            private _storeKey: string;
            get storeKey(): string {
                return this._storeKey;
            }
            set storeKey(value: string) {
                this._storeKey = value;
            }

            private _sharedSecret: string;
            get sharedSecret(): string {
                return this._sharedSecret;
            }
            set sharedSecret(value: string) {
                this._sharedSecret = value;
            }

            private _fiIdentifier: string;
            get fiIdentifier(): string {
                return this._fiIdentifier;
            }
            set fiIdentifier(value: string) {
                this._fiIdentifier = value;
            }

            private _adminServiceUrl: string;
            get adminServiceUrl(): string {
                return this._adminServiceUrl;
            }
            set adminServiceUrl(value: string) {
                this._adminServiceUrl = value;
            }

            private _serviceUrl: string;
            get serviceUrl(): string {
                return this._serviceUrl;
            }
            set serviceUrl(value: string) {
                this._serviceUrl = value;
            }

            private _vendorName: string;
            get vendorName(): string {
                return this._vendorName;
            }
            set vendorName(value: string) {
                this._vendorName = value;
            }

            private _androidAppBundleId: string;
            get androidAppBundleId(): string {
                return this._androidAppBundleId;
            }
            set androidAppBundleId(value: string) {
                this._androidAppBundleId = value;
            }

            private _iosAppBundleId: string;
            get iosAppBundleId(): string {
                return this._iosAppBundleId;
            }
            set iosAppBundleId(value: string) {
                this._iosAppBundleId = value;
            }

            private _syncAccountsEnabled: boolean;
            get syncAccountsEnabled(): boolean {
                return this._syncAccountsEnabled;
            }
            set syncAccountsEnabled(value: boolean) {
                this._syncAccountsEnabled = value;
            }


    constructor() {}


        toSettings(): Setting[] {
            return [
                { key: "ProfitStars.StoreId", value: this._storeId, dataType: 'number', label: "Store Id" },
                { key: "ProfitStars.EntityId", value: this._entityId, dataType: 'number', label: "Entity Id" },
                { key: "ProfitStars.StoreKey", value: this._storeKey, dataType: 'string', label: "Store Key" },
                { key: "ProfitStars.SharedSecret", value: this._sharedSecret, dataType: 'string', label: "Shared Secret" },
                { key: "ProfitStars.FiIdentifier", value: this._fiIdentifier, dataType: 'string', label: "Fi Identifier" },
                { key: "ProfitStars.AdminServiceUrl", value: this._adminServiceUrl, dataType: 'string', label: "Admin Service Url" },
                { key: "ProfitStars.ServiceUrl", value: this._serviceUrl, dataType: 'string', label: "Service Url" },
                { key: "ProfitStars.VendorName", value: this._vendorName, dataType: 'string', label: "Vendor Name" },
                { key: "ProfitStars.AndroidAppBundleId", value: this._androidAppBundleId, dataType: 'string', label: "Android App Bundle Id" },
                { key: "ProfitStars.IosAppBundleId", value: this._iosAppBundleId, dataType: 'string', label: "Ios App Bundle Id" },
                { key: "ProfitStars.SyncAccountsEnabled", value: this._syncAccountsEnabled, dataType: 'boolean', label: "Sync Accounts Enabled" },
            ];
        }

}